### PR TITLE
Preparing for release of the 3.0.0 version

### PR DESCRIPTION
**Pull request by egor-bryzgalov**
**Description:** Made in accordance with [#711](https://github.com/microsoft/azure-pipelines-task-lib/issues/711). 

Changes: `releases/3.x` branch is merged to master, conflicts are resolved and version is bumped.

Since most of the pipeline tasks are running `3.0.6-preview.x` version and no issues were found related to the task-lib update it was decided to release a stable 3.x version.

Please consider [the difference between this branch and releases/3.x.](https://github.com/microsoft/azure-pipelines-task-lib/compare/releases/3.x...releases/3.x-merging)

### CODE REVIEW

**Review by anatolybolshakov — APPROVED** [review]

**Review by DaniilShmelev — APPROVED** [review]

**Review by max-zaytsev — APPROVED** [review]

**Review by EzzhevNikita — APPROVED** [review]

File: node/README.md [hunk 1]
```diff
@@ -11,19 +11,31 @@ Cross platform tasks are written in TypeScript.  It is the preferred way to writ
...
 
 Documentation: [TypeScript API](docs/azure-pipelines-task-lib.md), [task JSON schema](https://aka.ms/vsts-tasks.schema.json)
 
 Guidance: [Finding Files](docs/findingfiles.md), [Minimum agent version](docs/minagent.md), [Proxy](docs/proxy.md), [Certificate](docs/cert.md)
 
+## Node 10 Upgrade Notice
+
+Azure DevOps is currently working to establish Node 10 as the new preferred runtime for tasks, upgrading from Node 6.
+Relevant work is happening in the releases/3.x branch and will be merged into master when we have successfully migrated the majority of our first party tasks.
+Any changes should be dual-commited to both `master` and `releases/3.x` until the merge into master occurs.
```
Comment (alex-peck) [inline node/README.md:24]: Remove references to `releases/3.x` branch?

**Review by alex-peck — APPROVED** [review]

**Review by damccorm — APPROVED** [review]

File: node/README.md (hunk 1 above)
Comment (egor-bryzgalov) [reply node/README.md]: Thank you for the review! Fixed it.

[the remark this answers is not in the document]
Comment (egor-bryzgalov) [reply node/README.md]: Thank you for the review! I've updated it.

File: node/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "azure-pipelines-task-lib",
-  "version": "2.12.2",
+  "version": "3.0.0",
```
[the remark this answers is not in the document]
Comment (egor-bryzgalov) [reply node/package.json]: Thank you for the point! I've changed it.

**Comment by egor-bryzgalov** [comment]
@damccorm Would you mind releasing the latest version of the package? Thank you!

**Comment by damccorm** [comment]
Yep, published!
________________________________

**Pull request merged**